src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs

```diff
@@ -411,71 +412,381 @@ public async Task AcceptToMemoryPool_WithSegWitValidTxns_IsSuccessfullAsync()
+            tx.AddInput(new TxIn(new OutPoint(new uint256(0), uint.MaxValue), PayToPubkeyHashTemplate.Instance.GenerateScriptPubKey(minerSecret.PubKey)));
+            tx.AddOutput(new TxOut(new Money(Money.CENT * 11), destSecret.PubKeyHash));
+            Assert.True(tx.IsCoinBase);
+
+            // It is not necessary to sign the transaction as we want the scriptSig left untouched and (2 < size < 100) bytes to pass the coinbase size check.
+            var state = new MempoolValidationState(false);
+            bool isSuccess = await validator.AcceptToMemoryPool(state, tx);
+
+            // Tests PreMempoolChecks context.Transaction.IsCoinBase
+            Assert.False(isSuccess, "Coinbase should not be accepted to mempool.");
+            Assert.Equal(MempoolErrors.Coinbase, state.Error);
         }
 
-        [Fact(Skip = "Not implemented yet.")]
-        public void AcceptToMemoryPool_TxIsNonStandardVersionUnsupported_ReturnsFalse()
+        [Fact]
+        public async void AcceptToMemoryPool_TxIsCoinbaseWithInvalidSize_ReturnsFalseAsync()
         {
-            // TODO:Test the cases in PreMempoolChecks CheckStandardTransaction
-            // - Check version number
+            string dataDir = GetTestDirectoryPath(this);
+
+            var minerSecret = new BitcoinSecret(new Key(), KnownNetworks.RegTest);
+            ITestChainContext context = await TestChainFactory.CreateAsync(KnownNetworks.RegTest, minerSecret.PubKey.Hash.ScriptPubKey, dataDir);
+            IMempoolValidator validator = context.MempoolValidator;
+            Assert.NotNull(validator);
+
+            var destSecret = new BitcoinSecret(new Key(), KnownNetworks.RegTest);
+            var tx = new Transaction();
+
+            // Create a transaction that looks like a coinbase. But the consensus rules that apply to coinbases will reject it anyway.
+            // We need two forms of the test because submitting an incorrectly constructed coinbase triggers a consensus rule instead
+            // of the coinbase being rejected by the mempool for being a coinbase.
+            // TODO: Instead of creating multiple versions of all such tests we should perhaps find a way of testing the applicable rules in the mempool simultaneously.
+            tx.AddInput(new TxIn(new OutPoint(new uint256(0), uint.MaxValue), PayToPubkeyHashTemplate.Instance.GenerateScriptPubKey(minerSecret.PubKey)));
+            tx.AddOutput(new TxOut(new Money(Money.CENT * 11), destSecret.PubKeyHash));
+            tx.Sign(KnownNetworks.RegTest, minerSecret, false);
+            Assert.True(tx.IsCoinBase);
+
+            var state = new MempoolValidationState(false);
+            bool isSuccess = await validator.AcceptToMemoryPool(state, tx);
+
+            // Tests PreMempoolChecks invokes CheckPowTransactionRule and is enforced for mempool transactions.
+            Assert.False(isSuccess, "Coinbase with incorrect size should not be accepted to mempool.");
```

Comment:
I think its essential to check the error itself that broke like the test above that did this ` Assert.Equal(MempoolErrors.Coinbase, state.Error);` if not you may get a false positive, test failed on the wrong rule.